src/modules/transaction/sagas.spec.ts

```diff
@@ -0,0 +1,222 @@
+        url: '',
+        replacedBy: null
+      }
+    })
+
+    describe('and the transaction becomes confirmed', () => {
+      test('should use fibonacci backoff until confirmation and fix the transaction', async () => {
+        const { hash } = transaction
+        const mockReceipt = { logs: [] }
+        const revertedTx = {
+          ...transaction,
+          status: TransactionStatus.REVERTED
+        }
+        const action = {
+          type: '[Request] Fetch Transaction' as const,
+          payload: {
+            hash,
+            address: '0x123',
+            action: {
+              type: 'SOME_ACTION',
+              payload: buildTransactionPayload(
+                transaction.chainId,
+                hash,
+                {},
+                transaction.chainId
+              )
+            }
+          }
+        }
+
+        return expectSaga(handleRegularTransactionRequest, action)
+          .provide([
+            [select(getTransactionInState, hash), revertedTx],
+            [
+              call(getTransactionFromChain, '0x123', transaction.chainId, hash),
+              {
+                ...revertedTx,
+                status: TransactionStatus.CONFIRMED,
+                receipt: mockReceipt
+              }
+            ]
+          ])
+          .withState({
+            transaction: {
+              data: [revertedTx],
+              loading: [],
+              error: null
+            },
+            wallet: {
+              data: {
+                address: '0x123'
+              }
+            }
+          })
+          .put(
+            fetchTransactionSuccess({
+              ...revertedTx,
+              status: TransactionStatus.CONFIRMED,
+              receipt: { logs: [] }
+            })
+          )
+          .run({ timeout: 15000 })
+      })
+    })
+  })
+
+  describe('when watching reverted transaction', () => {
+    let transaction: Transaction
+    let address: string
+    let hash: string
+    let chainId: ChainId
+
+    beforeEach(() => {
+      address = '0x123'
+      hash = '0x456'
+      chainId = ChainId.ETHEREUM_MAINNET
+      transaction = {
+        ...buildTransactionPayload(chainId, hash, {}, chainId),
+        events: [],
+        hash,
+        from: address,
+        chainId,
+        status: TransactionStatus.REVERTED,
+        timestamp: Date.now(),
+        nonce: 0,
+        withReceipt: false,
+        isCrossChain: false,
+        replacedBy: null,
+        actionType: 'SOME_ACTION',
+        url: ''
+      }
+    })
+
+    describe('and the transaction expires', () => {
+      it('should stop polling after expiration threshold', () => {
+        const expiredTransaction = {
+          ...transaction,
+          timestamp: Date.now() - 25 * 60 * 60 * 1000 // 25 hours ago
+        }
+
+        return expectSaga(handleWatchRevertedTransaction, {
+          type: WATCH_REVERTED_TRANSACTION,
+          payload: { hash }
+        })
```

Comment:
Would you mind using here the `watchRevertedTransaction` action?
```suggestion
        return expectSaga(handleWatchRevertedTransaction, watchRevertedTransaction(hash))
```